ExamenProeveUnity/Assets/Scripts/Runtime/Player/PlayerGrab.cs

```diff
@@ -22,18 +22,23 @@ private void Awake()
             _rigidbody ??= GetComponent<Rigidbody>();
         }
 
-        private void Update()
+        private void OnRightGrab(InputValue value)
         {
-            HandleMousePressed();
-            HandleMouseRelease();
+            if (mouseType != MouseType.Right) return;
+            if(value.isPressed) HandleMousePressed();
+            else HandleMouseRelease();
+        }
+        
+        private void OnLeftGrab(InputValue value)
+        {
+            if (mouseType != MouseType.Left) return;
+            if(value.isPressed) HandleMousePressed();
+            else HandleMouseRelease();
         }
 
         private void HandleMouseRelease()
```

Comment:
new input system uses more than mouse presses
change name to something like 'HandleHandRelease()'